apiserver/facades/client/action/interface.go

```diff
@@ -59,9 +67,67 @@ func (s *stateShim) FindEntity(tag names.Tag) (state.Entity, error) {
 	return s.st.FindEntity(tag)
 }
 
-func (s *stateShim) Model() (Model, error) {
-	return s.st.Model()
-}
 func (s *stateShim) WatchActionLogs(actionId string) state.StringsWatcher {
 	return s.st.WatchActionLogs(actionId)
 }
+
+// Methods moved from Model to State - implemented by getting the model first
+func (s *stateShim) AddAction(receiver state.ActionReceiver, operationID, name string, payload map[string]interface{}, parallel *bool, executionGroup *string) (state.Action, error) {
```

Comment:
Hey @leyao-daily,

I suspect there is some confusion here. We don't need to remove state model as part of this work. That will happen over time as we naturally move all of the things in model over to Dqlite.

Model as a concept in the database encompasses not just the information about the model such as its name and type but also things like all the applications and machines in the model. This task is just about moving what we can about the model's information away from Mongo. These are things like reading the model's type, name and uuid.

You will see in the Jira task there is a list of things that we are targeting as part of this PR :).